### PR TITLE
"Minio" (S3 Compatible) Integration docs

### PR DESCRIPTION
**Pull request by dylangrandmont**
### Linked issue(s):

Documentation support for https://linear.app/kolena/issue/PRO-529/guided-bucket-connection-flow

### What change does this PR introduce and why?

Introduce first pass of documentation for Cloud Storage Integrations. Here a new section under "Advanced Usage" is introduced containing instructions for adding "minIO" (really, any s3-compatible) integrations

### CODE REVIEW

File: docs/advanced-usage/connecting-cloud-storage/s3-compatible.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+---
+
+Integrations can be established to S3-compatible systems (for example, MinIO).
+To get started, ensure you have admin access within Kolena.
+Navigate to your [Organization Settings's Integration Tab](https://app.kolena.io/redirect/organization?tab=integrations) and click "Add Integration", then "MinIO".
+
+Steps performed outside of Kolena are shown for a subset of possible S3-compatible systems.
+You may need to consult documentation for your provider to perform equivalent steps.
+
+#### 1. Create a Service User for Kolena
```
Comment (gordonhart) [inline docs/advanced-usage/connecting-cloud-storage/s3-compatible.md]: I would use larger headings here, maybe `###`?

File: docs/advanced-usage/connecting-cloud-storage/s3-compatible.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+=== "`minIO`"
```
Comment (gordonhart) [inline docs/advanced-usage/connecting-cloud-storage/s3-compatible.md]: Should this be "MinIO" or `MinIO`? Not sure what the preferred style is, but we should find it and stick to it

File: docs/advanced-usage/connecting-cloud-storage/s3-compatible.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+---
+subtitle: MinIO, Oracle, Hitachi
+---
+
+Integrations can be established to S3-compatible systems (for example, MinIO).
+To get started, ensure you have admin access within Kolena.
+Navigate to your [Organization Settings's Integration Tab](https://app.kolena.io/redirect/organization?tab=integrations) and click "Add Integration", then "MinIO".
+
+Steps performed outside of Kolena are shown for a subset of possible S3-compatible systems.
+You may need to consult documentation for your provider to perform equivalent steps.
```
Comment (gordonhart) [inline docs/advanced-usage/connecting-cloud-storage/s3-compatible.md:14]: We should include a list up here detailing the known-good S3-compatible providers and linking to their documentation.

File: docs/advanced-usage/connecting-cloud-storage/s3-compatible.md (hunk 2 above)
Comment (dylangrandmont) [reply docs/advanced-usage/connecting-cloud-storage/s3-compatible.md]: I'll stick with `MinIO` as this appears to be their preferred branding

File: docs/advanced-usage/connecting-cloud-storage/s3-compatible.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+---
```
Comment (yifanwu-kolena) [inline docs/advanced-usage/connecting-cloud-storage/s3-compatible.md:1]: we might need to emphasize that the bucket name needs to respect [s3 bucket naming rules](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html) due to strict minio checking: e.g. `_` is invalid

File: mkdocs.yml [hunk 5]
```diff
@@ -118,7 +121,8 @@ markdown_extensions:
   - pymdownx.superfences
   - pymdownx.tabbed:
       alternate_style: true
-      slugify: !!python/object/apply:pymdownx.slugs.slugify  # NOTE: insiders only
+      slugify:
+        !!python/object/apply:pymdownx.slugs.slugify # NOTE: insiders only
```
Comment (gordonhart) [inline mkdocs.yml]: Double-checking: does this change any functionality?

File: docs/overrides/.icons/kolena/cloud-16.svg
Comment (gordonhart) [inline docs/overrides/.icons/kolena/cloud-16.svg:1]: Would prefer to add both the 16 and the 20 — we have both versions of all icons currently.

**Review by gordonhart — APPROVED** [review]

File: mkdocs.yml (hunk 5 above)
Comment (dylangrandmont) [reply mkdocs.yml]: Reverting in case

**Pull request merged**